src/components/Email/ManageWalletModalWithEmail.tsx

```diff
@@ -0,0 +1,193 @@
+import { HStack, VStack } from '~/components/core/Spacer/Stack';
+import { BaseM, TitleDiatypeL, TitleDiatypeM } from '~/components/core/Text/Text';
+import Toggle from '~/components/core/Toggle/Toggle';
+import EmailManager from '~/components/Email/EmailManager';
+import ManageWallets from '~/components/ManageWallets/ManageWallets';
+import { useReportError } from '~/contexts/errorReporting/ErrorReportingContext';
+import { useToastActions } from '~/contexts/toast/ToastContext';
+import { ManageWalletModalWithEmailFragment$key } from '~/generated/ManageWalletModalWithEmailFragment.graphql';
+
+import useUpdateEmailNotificationSettings from './useUpdateEmailNotificationSettings';
+
+type Props = {
+  queryRef: ManageWalletModalWithEmailFragment$key;
+  newAddress?: string;
+  onEthAddWalletSuccess?: () => void;
+  onTezosAddWalletSuccess?: () => void;
+};
+
+function ManageWalletsModalWithEmail({
+  newAddress,
+  queryRef,
+  onEthAddWalletSuccess,
+  onTezosAddWalletSuccess,
+}: Props) {
+  const query = useFragment(
+    graphql`
+      fragment ManageWalletModalWithEmailFragment on Query {
+        viewer @required(action: THROW) {
+          ... on Viewer {
+            email @required(action: THROW) {
+              email
+              emailNotificationSettings {
+                unsubscribedFromAll
+                unsubscribedFromNotifications
+              }
+            }
+          }
+        }
+
+        ...EmailManagerFragment
+        ...ManageWalletsFragment
+      }
+    `,
+    queryRef
+  );
+
+  const updateEmailNotificationSettings = useUpdateEmailNotificationSettings();
+  const [isEmailNotificationChecked, setIsEmailNotificationChecked] = useState(false);
+  const [isShowAddEmail, setIsShowAddEmail] = useState(false);
+  const { pushToast } = useToastActions();
+  const reportError = useReportError();
+
+  // Invert the value since in the db stored as unsubscribe while in the UI it's subscribe
+  const isEmailNotificationSubscribed =
+    query?.viewer?.email?.emailNotificationSettings?.unsubscribedFromNotifications || false;
+
+  const isEmailUnsubscribedFromAll =
+    !!query?.viewer?.email?.emailNotificationSettings?.unsubscribedFromAll ?? false;
+
+  // Set the initial state of the email notification toggle
+  // so we can instantly toggle it on/off without waiting for the mutation to complete
+  useEffect(() => {
+    setIsEmailNotificationChecked(isEmailNotificationSubscribed);
+  }, [isEmailNotificationSubscribed]);
+
+  // If the user already have email attached, toggle the email manager ui
+  const userEmail = query?.viewer?.email?.email;
+  useEffect(() => {
+    setIsShowAddEmail(Boolean(userEmail));
+  }, [userEmail]);
+
+  const [isPending, setIsPending] = useState(false);
+
+  const handleEmailNotificationChange = useCallback(
+    async (checked: boolean) => {
+      const unsubscribedFromNotifications = !checked;
+      setIsEmailNotificationChecked(checked);
+      setIsPending(true);
```

Comment:
`usePromisifiedMutation` returns `[submit, boolean]`. can we thread that through here so we don't have to manage that state ourselves?